aioredis/commands/cluster.py

```diff
@@ -7,94 +9,281 @@ class ClusterCommandsMixin:
         """Set the configuration epoch in a new node."""
-        fut = self._conn.execute(b'CLUSTER', b'SET-CONFIG-EPOCH', config_epoch)
+
+        try:
+            config_epoch = int(config_epoch)
+        except ValueError:
+            raise TypeError(
+                "Expected slot to be of type int, got {}".format(
+                    type(config_epoch)
+                )
+            )
+
+        fut = self.execute(b'CLUSTER', b'SET-CONFIG-EPOCH', config_epoch)
         return wait_ok(fut)
 
-    def cluster_setslot(self, slot, command, node_id):
+    def cluster_setslot(self, slot, command, node_id=None):
         """Bind a hash slot to specified node."""
-        pass    # TODO: Implement
+
+        try:
+            slot = int(slot)
+        except ValueError:
+            raise TypeError(
+                "Expected slot to be of type int, got {}".format(type(slot))
+            )
+
+        if command.upper() in self.SLOT_MIGRATING_STATES and \
+                node_id is not None:
+            fut = self.execute(b'CLUSTER', b'SETSLOT', slot, command, node_id)
+        elif command.upper() == self.SLOT_STABLE_STATE:
+            fut = self.execute(b'CLUSTER', b'SETSLOT', slot, command)
+        else:
+            raise ValueError('Invalid slot state: {0}'.format(command))
+
+        return wait_ok(fut)
 
     def cluster_slaves(self, node_id):
         """List slave nodes of the specified master node."""
-        pass    # TODO: Implement
+        fut = self.execute(b'CLUSTER', b'SLAVES', node_id)
+        return wait_convert(
+            fut, parse_cluster_nodes_lines, encoding=self.encoding
+        )
 
     def cluster_slots(self):
         """Get array of Cluster slot to node mappings."""
-        pass    # TODO: Implement
+        fut = self.execute(b'CLUSTER', b'SLOTS')
+        return wait_convert(fut, parse_cluster_slots)
+
+    def cluster_readonly(self):
+        """
+        Enables read queries for a connection to a Redis Cluster slave node.
+        """
+        fut = self.execute(b'READONLY')
+        return wait_ok(fut)
+
+    def cluster_readwrite(self):
+        """
+        Disables read queries for a connection to a Redis Cluster slave node.
+        """
+        fut = self.execute(b'READWRITE')
+        return wait_ok(fut)
+
+
+def _decode(s, encoding):
+    if encoding:
+        return decode(s, encoding)
+    return s
+
+
+def parse_info(resp, **options):
+    e = options.get('encoding')
+
+    lines = resp.strip().splitlines()
+    result = list(map(lambda s: s.split(_decode(b':', e)), lines))
+    return dict(result)
+
+
+def parse_node_slots(raw_slots, encoding=None):
+    """
+    @see: https://redis.io/commands/cluster-nodes#serialization-format
+    @see: https://redis.io/commands/cluster-nodes#special-slot-entries
+    """
+
+    slots, migrations = [], []
+    r_delimiter = _decode(b'->-', encoding)
+    l_delimiter = _decode(b'-<-', encoding)
+    m_delimiter = _decode(b'-', encoding)
```

Comment:
Better names would be migration_delimiter, import_delimiter and range_delimiter.